src/main/java/org/nibor/autolink/internal/WwwUrlScanner.java

```diff
@@ -0,0 +1,42 @@
+package org.nibor.autolink.internal;
+
+import org.nibor.autolink.LinkSpan;
+import org.nibor.autolink.LinkType;
+
+/**
+ * Scan for URLs starting from the trigger character "w", requires "www.".
+ * <p>
+ * Based on RFC 3986.
+ */
+public class WwwUrlScanner extends UrlScanner {
+
+    @Override
+    public LinkSpan scan(final CharSequence input, int triggerIndex, int rewindIndex) {
+        int afterDot = triggerIndex + 4;
+        if (afterDot >= input.length() || input.charAt(triggerIndex + 1) != 'w' || input.charAt(triggerIndex + 2) != 'w' || input.charAt(triggerIndex + 3) != '.') {
+            return null;
+        }
+
+        int first = triggerIndex;
```

Comment:
Actually, I am not quite convinced that would be a valid URL.
If that ought to be valid, would it not be along with ```http://``` ?